src/Parser/Lexer/Common.idr

```diff
@@ -45,3 +52,37 @@ ident : Flavour -> Lexer
 ident flavour =
   (pred $ isIdentStart flavour) <+>
     (many . pred $ isIdentTrailing flavour)
+
+export
+isIdentNormal : String -> Bool
+isIdentNormal = isIdent Normal
+
+export
+identNormal : Lexer
+identNormal = ident Normal
+
+export
+identAllowDashes : Lexer
+identAllowDashes = ident AllowDashes
+
+namespaceIdent' : Bool -> Lexer
+namespaceIdent' b = ident Capitalised <+> many leavingLastIdent
+  where
+    leavingLastIdent : Lexer
+    leavingLastIdent = if b then (is '.' <+> ident Capitalised)
+                       else (is '.' <+> ident Capitalised <+> expect (is '.'))
```

Comment:
What's the meaning of this `Bool` flag? Could you add a comment or maybe rename `b`?

Reply:
Hi @ziman a comment has been added.

Reply:
> How does your lexer disambiguate them?

It doesn't, it uses it in a different lexer for packages